go-controller/pkg/ovn/port_cache.go

```diff
@@ -0,0 +1,86 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+)
+
+type portCache struct {
+	sync.Mutex
+
+	cache map[string]*lpInfo
+}
+
+type lpInfo struct {
+	name          string
+	uuid          string
+	logicalSwitch string
+	ip            *net.IPNet
+	mac           net.HardwareAddr
+
+	// removeTime marks the time after which this port should be removed
+	// from the cache
+	removeTime time.Time
+}
+
+func newPortCache(stopChan <-chan struct{}) *portCache {
+	c := &portCache{
+		cache: make(map[string]*lpInfo),
+	}
+
+	// Removal must be deferred since, for example, NetworkPolicy pod handlers
+	// may run after the main pod handler and look for items in the cache
+	// on delete events.
```

Comment:
@dcbw I don't think this will scale with 1000s of logical switch ports.

Reply:
@danwinship the alternative is reference counting stuff in the port cache :( But this also may be temporary since it's only really required because of our multi-level informer setup right now, which I'm working on removing.

Reply:
@girishmg can you elaborate? we already have two separate per-LSP caches...

Reply:
@dcbw in our cluster we are going to have around 5000 logicals switch ports (aka 5000 pods). Now, with the deferred deletion strategy we will have to go through each of the 5000 ports when one pod gets deleted. Also, I don't know if we can assume that 10 second is going to be sufficient time for the network policy handler to run before we delete the entry from the cache.

Reply:
@girishmg updated to do a per-pod deferred deletion, since it seems very unlikely we'll be deleting 5000 pods at the same time.  Also bumped the timeout to a minute, hopefully that's OK until we can rework some of the network policy code to address Dan's comment above.